client/state/sites/reducer.js

```diff
@@ -105,6 +105,9 @@ export function items( state = {}, action ) {
 		case SITE_DELETE_RECEIVE:
 			return omit( state, action.siteId );
 
+		case SITE_REQUEST_FAILURE:
```

Comment:
It feels a little odd to remove a site if it fails to be loaded. What if we have it in state already, but the request fails because of timeout or a temporary server overload? Do we want to remove that site in that case?

In case you prefer to keep this behavior, you can combine the similar actions that omit a site:

```
case SITE_DELETE_RECEIVE:
case SITE_REQUEST_FAILURE:
case JETPACK_DISCONNECT_RECEIVE:
	return omit( state, action.siteId );
```